src/components/course/data/service.js

```diff
@@ -29,8 +30,24 @@ export default class CourseService {
     ])
       .then((responses) => responses.map(response => response.data));
 
+    // Check for the course_run_key URL param and remove all other course run data
+    // if the given course run key is for an available course run.
+    const courseDetails = camelCaseObject(data[0]);
+    const { courseRunKey } = camelCaseObject(qs.parse(window.location.search));
+    if (courseRunKey) {
+      const availableCourseRuns = getAvailableCourseRuns(courseDetails);
+      const availableCourseRunKeys = availableCourseRuns.map(({ key }) => key);
+      if (availableCourseRunKeys.includes(courseRunKey)) {
+        const { courseRuns } = courseDetails;
+        courseDetails.canonicalCourseRunKey = courseRunKey;
+        courseDetails.courseRunKeys = [courseRunKey];
+        courseDetails.courseRuns = courseRuns.filter(obj => obj.key === courseRunKey);
```

Comment:
Should this filter be on the `availableCourseRuns` to make sure we're only looking at enrollable runs?